src/EPR.Calculator.Service.Common/AzureSynapse/AzureSynapseRunner.cs

```diff
@@ -1,137 +1,156 @@
+        /// <param name="pipelineClientFactory">A factory that initializes pipeline clients
         /// (or generates mock clients when unit testing).</param>
+        /// <param name="logger">The logger instance to use for logging.</param>
         internal AzureSynapseRunner(
-            PipelineClientFactory pipelineClientFactory)
-            => this.PipelineClientFactory = pipelineClientFactory;
+            IPipelineClientFactory pipelineClientFactory, ILogger<AzureSynapseRunner> logger)
+        {
+            this.PipelineClientFactory = pipelineClientFactory;
+            this.logger = logger;
+        }
 
-        private PipelineClientFactory PipelineClientFactory { get; }
+        private IPipelineClientFactory PipelineClientFactory { get; }
 
-        /// <inheritdoc/>
+        /// <summary>
+        /// Processes the Azure Synapse Runner with the specified parameters.
+        /// </summary>
+        /// <param name="args">The parameters required to run the Azure Synapse Runner.</param>
+        /// <returns>A task that represents the asynchronous operation. The task result contains a boolean value indicating whether the pipeline run succeeded.</returns>
         public async Task<bool> Process(AzureSynapseRunnerParameters args)
         {
-            // instead of year will get financial year need to map finacial year to calendar year
+            this.logger.LogInformation("Azure synapse trigger process started");
+
+            // Instead of year, get financial year and map financial year to calendar year.
             // Trigger the pipeline.
             Guid pipelineRunId;
             pipelineRunId = await StartPipelineRun(
                 this.PipelineClientFactory,
                 args.PipelineUrl,
                 args.PipelineName,
-                Util.GetCalendarYear(args.FinancialYear));
+                args.FinancialYear);
+
+            this.logger.LogInformation($"pipelineRunId: {pipelineRunId}");
 
-            // Periodically check the status of pipeline until it's completed.
+            // Periodically check the status of the pipeline until it's completed.
             var checkCount = 0;
             var pipelineStatus = nameof(PipelineStatus.NotStarted);
+            var maxCheckCount = args.MaxCheckCount;
             do
             {
                 checkCount++;
                 try
                 {
+                    this.logger.LogInformation($"pipelineRunId checkCount: {checkCount}");
                     pipelineStatus = await GetPipelineRunStatus(
                         this.PipelineClientFactory,
                         args.PipelineUrl,
                         pipelineRunId);
-                    if (pipelineStatus != nameof(PipelineStatus.InProgress))
+
+                    this.logger.LogInformation($"pipelineStatus for pipelineRunId {pipelineRunId}: {pipelineStatus}");
+                    if (pipelineStatus.Contains("Succeeded"))
```

Comment:
Can you please use a constant for "succeeded"

Reply:
Resolved in PR#16